packages/desktop-client/src/components/common/Modal.tsx

```diff
@@ -100,69 +96,81 @@ export const Modal = ({
-            className={css(styles.lightScrollbar)}
-            style={{
-              outline: 'none', // remove focus outline
-            }}
-          >
-            <ModalContentContainer
-              noAnimation={noAnimation}
-              isActive={isActive(name)}
-              {...containerProps}
+      {/* A container for positioning the modal relative to the visual viewport */}
+      <div
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+          height: 'var(--visual-viewport-height)',
+          overflowY: 'auto',
+          transition: 'height .25s',
+        }}
+      >
+        <ReactAriaModal>
+          {modalProps => (
+            <Dialog
+              aria-label={t('Modal dialog')}
+              className={css(styles.lightScrollbar)}
               style={{
-                flex: 1,
-                padding: 10,
-                willChange: 'opacity, transform',
-                maxWidth: '90vw',
-                minWidth: '90vw',
-                maxHeight: '90vh',
-                minHeight: 0,
-                borderRadius: 6,
-                //border: '1px solid ' + theme.modalBorder,
-                color: theme.pageText,
-                backgroundColor: theme.modalBackground,
-                opacity: isHidden ? 0 : 1,
-                [`@media (min-width: ${tokens.breakpoint_small})`]: {
-                  minWidth: tokens.breakpoint_small,
-                },
-                overflowY: 'auto',
-                ...styles.shadowLarge,
-                ...containerProps?.style,
+                outline: 'none', // remove focus outline
               }}
             >
-              <View style={{ paddingTop: 0, flex: 1, flexShrink: 0 }}>
-                {typeof children === 'function'
-                  ? children(modalProps)
-                  : children}
-              </View>
-              {isLoading && (
-                <View
-                  style={{
-                    position: 'absolute',
-                    top: 0,
-                    left: 0,
-                    right: 0,
-                    bottom: 0,
-                    backgroundColor: theme.pageBackground,
-                    alignItems: 'center',
-                    justifyContent: 'center',
-                    zIndex: 1000,
-                  }}
-                >
-                  <AnimatedLoading
-                    style={{ width: 20, height: 20 }}
-                    color={theme.pageText}
-                  />
+              <ModalContentContainer
+                noAnimation={noAnimation}
+                isActive={isActive(name)}
+                {...containerProps}
+                style={{
+                  flex: 1,
+                  padding: 10,
+                  willChange: 'opacity, transform',
+                  maxWidth: '90vw',
+                  minWidth: '90vw',
+                  maxHeight: 'var(--visual-viewport-height)',
```

Comment:
This makes some of the taller modals look a bit odd on desktop, could you limit it to expanding on mobile only?
```suggestion
                  maxHeight: isNarrowWidth ? 'var(--visual-viewport-height)' : '90vh',
```

Reply:
Ah yes, missed that, changed it to `calc(var(--visual-viewport-height) * 0.9)` which should be effectively the same (90% of the viewport height): https://github.com/actualbudget/actual/pull/4427/commits/b56ffa7fb1e0625822aa12f42e4088193364db8a